cmd/rpc-server_test.go

```diff
@@ -43,21 +46,40 @@ func (t *Arith) Add(args ArithArgs, reply *ArithReply) error {
 
 func TestGoHTTPRPC(t *testing.T) {
 	newServer := newRPCServer()
-	newServer.Register(new(Arith))
+	newServer.Register(&Arith{
+		AuthRPCServer: AuthRPCServer{},
+	})
 
 	mux := router.NewRouter().SkipClean(true)
 	mux.Path("/foo").Handler(newServer)
 
 	httpServer := httptest.NewServer(mux)
 	defer httpServer.Close()
 
-	client := newRPCClient(httpServer.Listener.Addr().String(), "/foo", false)
+	rootPath, err := newTestConfig("us-east-1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(rootPath)
+
+	creds := globalServerConfig.GetCredential()
+	client := newAuthRPCClient(authConfig{
+		serverAddr:      httpServer.Listener.Addr().String(),
+		serviceName:     "Arith",
+		serviceEndpoint: "/foo",
+		accessKey:       creds.AccessKey,
+		secretKey:       creds.SecretKey,
+	})
 	defer client.Close()
 
+	if err = client.Login(); err != nil {
+		t.Fatal(err)
+	}
+
 	// Synchronous calls
 	args := &ArithArgs{7, 8}
 	reply := new(ArithReply)
-	if err := client.Call("Arith.Add", args, reply); err != nil {
+	if err = client.rpcClient.Call("Arith.Add", args, reply); err != nil {
```

Comment:
Shouldn't client.Call() also work ?

Reply:
That would cause dead lock. Login is the first call where dial happens. This function is called by Call()